src/model/core/co2.jl

```diff
@@ -0,0 +1,121 @@
+@doc raw""" 
+    co2!(EP::Model, inputs::Dict)
+
+This function creates expressions to account for CO2 emissions as well as captured and sequestrated 
+CO2 from thermal generators. It also has the capability to model the negative CO2 emissions 
+from bioenergy with carbon capture and storage. 
+
+***** Expressions *****
+
+For thermal generators which combust fuels (e.g., coal, natural gas, and biomass),
+the net CO2 emission to the environment is a function of fuel consumption, CO2 capture fraction, and whether the feedstock is biomass. 
+The combustion of biomass (e.g., from wastes or agriculture residues) is typically considered to be 
```

Comment:
I know we don't usually track upstream emissions from fuel production, but there are many researchers who would strongly object to this statement. Depending on the source, changes to soil carbon, etc, biomass can be far from carbon-neutral.

Maybe a larger change is needed to account for upstream emissions from each fuel type. Methane from gas/coal extraction and land use change from biomass. Tracking these emissions outside the fenceline would allow for more flexible accounting and tracking emissions as part of the whole economy.